### DISCUSSION TITLE
Frequently getting "504 Gateway Time-out - The server didn't respond in time." Errors

### DISCUSSION THREAD

**Discussion by counselinginsight** [deepgram/community#617] [General help]
### Which Deepgram product are you using?

Deepgram API

### Details

I've been frequently getting 504 Gateway Time-out errors from deepgram.

Heres a snippet of the code I'm using for deepgram:

Request ids:

cbb5480e-dccc-4198-9c2e-80731529a6ed
96144c67-1ea9-466d-80f0-b0234d2ac8cf
c03e72b2-08c2-4589-a76a-d18b0e8d5f53

In the output logs, they still show as pending.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

POST /v1/listen?model=nova-2&filler_words=true&diarize=true&smart_format=true&detect_entities=true&detect_language=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

c03e72b2-08c2-4589-a76a-d18b0e8d5f53

### If possible, please attach your code or paste it into the text box.

```
const deepgram = new Deepgram(DEEPGRAM_API_KEY)
response = await deepgram.transcription.preRecorded(
  {
    buffer: file,
    mimetype,
  },
  {
    model,
    filler_words: transcribe.type !== 'dictation',
    diarize: transcribe.type !== 'dictation',
    smart_format: true,
    detect_entities: transcribe.type !== 'dictation',
    detect_language: true,
  }
)
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @counselinginsight, I've taken a look at these request IDs you provided. I see your file is around 2 hours long, which is on the longer side, and you're also using `detect_entities=true`. The entity detection feature is more compute-intensive, and paired with the longer audio file, the request is taking a long time, and we're not handling it elegantly.

When I remove the `detect_entities=true` param and replicate your request, I get a successful response in 56 seconds. Running your request as-is, I still see it going at 8 minutes and counting. So I recommend only using that parameter for shorter audio files.

**Reply by counselinginsight** [reply]
Thanks for the response.

On our end, we are getting this message:
```
DG: {"error":"<html><body><h1>504 Gateway Time-out</h1>\nThe server didn't respond in time.\n</body></html>\n"} 
```

So the requests are dropping. We need to include detect entities right now in our system.
Whats the maximum processing time that deepgram supports? Is there a way to increase the timeout on our end?

**Reply by jkroll-deepgram** [reply]
That looks like an SDK error - can you share which SDK and version you're using? There may be a way to configure the timeout further. 

We don't have an official length of audio on which `detect_entities` is supported, but you could take a quick binary search approach on what response speed/success you're seeing on different sub-lengths of audio. I'm also investigating further on my end about reasonable latency/audio length expectations.

**Reply by counselinginsight** [reply]
We are using the node api sdk from npm. `@deepgram/sdk`

**Reply by lukeocodes** [reply]
@jkroll-deepgram that is indeed the output of the old version of the SDK, so this isn't related to the other fetch issue.
